src/cpu/registers.rs

```diff
@@ -0,0 +1,280 @@
+// pub const BOOTROM_START_ADDR: u16 = 0x0000;
+// pub const TOP_OF_STACK_ADDR: u16 = 0xFFFE;
+
+const ZERO_FLAG: u8 = 7;
+const SUBTRACT_FLAG: u8 = 6;
+const HALF_CARRY_FLAG: u8 = 5;
+const CARRY_FLAG: u8 = 4;
+
+// 8-bit registers
+#[derive(Clone, Copy, Debug)]
+pub enum R8 {
+    A,
+    B,
+    C,
+    D,
+    E,
+    F,
+    H,
+    L,
+}
+
+// 16-bit registers
+#[derive(Clone, Copy, Debug)]
+pub enum R16 {
+    AF,
+    BC,
+    DE,
+    HL,
+    SP,
+    PC,
+}
+
+// F-register flags
+pub enum Flag {
+    Z,
+    N,
+    H,
+    C,
+}
+
+pub struct Registers {
+    a: u8,
+    b: u8,
+    c: u8,
+    d: u8,
+    e: u8,
+    f: Flags,
+    h: u8,
+    l: u8,
+    sp: u16, // Stack pointer
+    pc: u16, // Program counter
+}
+
+impl Registers {
+    pub fn new() -> Self {
+        Registers {
+            a: 0,
+            b: 0,
+            c: 0,
+            d: 0,
+            e: 0,
+            f: 0.into(),
+            h: 0,
+            l: 0,
+            sp: 0,
+            pc: 0,
+        }
+    }
+
+    pub fn get16(&self, register: R16) -> u16 {
+        let (high_register, low_register) = match register {
+            R16::AF => (R8::A, R8::F),
+            R16::BC => (R8::B, R8::C),
+            R16::DE => (R8::D, R8::E),
+            R16::HL => (R8::H, R8::L),
+            R16::SP => return self.sp,
+            R16::PC => return self.pc,
+        };
+
+        let high = self.get(high_register);
+        let low = self.get(low_register);
+
+        let mut word: u16 = 0;
```

Comment:
Now that you've presumably debugged it, these four lines read a bit awkward. I would have rewritten this stanza as just the return value:

```
(high as u16) <<8  | (low as u16)
```